### ISSUE TITLE
ENH: Force positive EDSModel components

### ISSUE THREAD

**Issue by vidartf** [hyperspy/hyperspy#884]
It would make sense if the default fitting of `EDSModel` components (i.e. the Gaussians) were bounded to be positive. For this to be the case `EDSModel.fit()` should switch the default values of its keyword args to `fitter='mpfit'` and `bounded=True`.

**Comment by dnjohnstone** [comment]
This has got to be sensible - just requires switching those keyword defaults right, or is it any more subtle?

So shall one of us just do it?

**Comment by francisco-dlp** [comment]
Easy and pretty useful, +1000

**Comment by dnjohnstone** [comment]
Ok - see #946

**Comment by tjof2** [comment]
Closing since #946 was closed and no one looks to have complained in 3+ years. Besides, leaving the decision to bound parameters up to the user is probably better.

**Comment by ericpre** [comment]
Well, I still think this would be a good thing to revisit at some point, particularly that since then, there have been some progress with respect to fitting - at the time, the issue with #946 is that the tests were failing.

**Comment by tjof2** [comment]
Good point @ericpre - reopened. It may also be easier I think after #2441 since mpfit won't be needed (i.e. leastsq will support bounds properly).

**Comment by tjof2** [comment]
Actually could even tackle it as part of #2441, what do you think?

**Comment by ericpre** [comment]
#2441 is already quite big, but that should be fine.

**Comment by tjof2** [comment]
Let's do it separately
